### PR TITLE
[onewire] merge OwserverBridgeHandler into OwBaseBridgeHandler

### PR DESCRIPTION
**Pull request by J-N-K**
This PR merges the implementation of the owserver bridge handler with the abstract bridge handler defintion. With increasing complexity of the binding (e.g. extended device id support) alternative bridge handler implementation are unlikely as they would require a JAVA rewrite of most of the owfs code.

This merge allows a lot of simplification of the code in other places, reducing the possibility of future bugs and simplifing the implementation of other devices.

- [x] merge code
- [x] remove OwDeviceParameterMap
- [x] testing

Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

**Comment by J-N-K** [comment]
Anything missing here from my side?

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java [hunk 1]
```diff
@@ -143,46 +249,108 @@ public OwSensorType getType(SensorId sensorId) throws OwException {
...
-            owserverConnection.writeDecimalType(
-                    ((OwserverDeviceParameter) parameter.get(THING_TYPE_OWSERVER)).getPath(sensorId), value);
+            owserverConnection.writeDecimalType(parameter.getPath(sensorId), value);
         }
     }
 
+    /**
+     * writes a BitSet to the sensor
+     *
+     * @param sensorId sensorId the sensor's full ID
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java]: Double "sensorId".

```suggestion
     * @param sensorId the sensor's full ID
```

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java [hunk 2]
```diff
@@ -143,46 +249,108 @@ public OwSensorType getType(SensorId sensorId) throws OwException {
...
+            return owserverConnection.readString(parameter.getPath(sensorId));
         }
     }
 
-    @Override
-    public void writeDecimalType(SensorId sensorId, OwDeviceParameterMap parameter, DecimalType value)
+    /**
+     * writes a DecimalType to the sensor
+     *
+     * @param sensorId sensorId the sensor's full ID
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java]: Double "sensorId".

```suggestion
     * @param sensorId the sensor's full ID
```

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java [hunk 3]
```diff
@@ -143,46 +249,108 @@ public OwSensorType getType(SensorId sensorId) throws OwException {
...
+            return owserverConnection.readDecimalTypeArray(parameter.getPath(sensorId));
         }
     }
 
-    @Override
-    public String readString(SensorId sensorId, OwDeviceParameterMap parameter) throws OwException {
+    /**
+     * read a string from a sensor
+     *
+     * @param sensorId sensorId the sensor's full ID
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java]: Double "sensorId".

```suggestion
     * @param sensorId the sensor's full ID
```

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java [hunk 4]
```diff
@@ -143,46 +249,108 @@ public OwSensorType getType(SensorId sensorId) throws OwException {
...
+     * @throws OwException
+     */
+    public BitSet readBitSet(SensorId sensorId, OwserverDeviceParameter parameter) throws OwException {
+        return BitSet.valueOf(new long[] { ((DecimalType) readDecimalType(sensorId, parameter)).longValue() });
+    }
+
+    /**
+     * read an array of decimal values from a sensor
+     *
+     * @param sensorId sensorId the sensor's full ID
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java]: Double "sensorId".

```suggestion
     * @param sensorId the sensor's full ID
```

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java [hunk 5]
```diff
@@ -143,46 +249,108 @@ public OwSensorType getType(SensorId sensorId) throws OwException {
...
+            return owserverConnection.readDecimalType(parameter.getPath(sensorId));
         }
     }
 
-    @Override
-    public OwPageBuffer readPages(SensorId sensorId) throws OwException {
+    /**
+     * read a BitSet value from a sensor
+     *
+     * @param sensorId sensorId the sensor's full ID
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java]: Double "sensorId".

```suggestion
     * @param sensorId the sensor's full ID
```

File: addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java [hunk 6]
```diff
@@ -46,17 +53,25 @@
...
-public class OwserverBridgeHandler extends OwBaseBridgeHandler {
+public class OwserverBridgeHandler extends BaseBridgeHandler {
     public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_OWSERVER);
 
     private final Logger logger = LoggerFactory.getLogger(OwserverBridgeHandler.class);
 
+    // set by implementation when bridge is ready
+    protected boolean refreshable = false;
+
+    protected ScheduledFuture<?> refreshTask = scheduler.scheduleWithFixedDelay(() -> refresh(), 1, 1000,
```
[the remark this answers is not in the document]
Comment (J-N-K) [reply addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwserverBridgeHandler.java]: Nice. Didn't know that works.

**Pull request merged**